In [56]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from pathlib import Path
import os

In [32]:
#전체데이터 가져오기
Product = pd.read_csv("D:/Digital_Trend_Analyzer/Product.csv")
Search1 = pd.read_csv("D:/Digital_Trend_Analyzer/Search1.csv")
Search2 = pd.read_csv("D:/Digital_Trend_Analyzer/Search2.csv")
Custom = pd.read_csv("D:/Digital_Trend_Analyzer/Custom.csv")
Session = pd.read_csv("D:/Digital_Trend_Analyzer/Session.csv")
Master = pd.read_csv("D:/Digital_Trend_Analyzer/Master.csv")

C:\Users\ASUS\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3020: DtypeWarning: Columns (7) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


## Checking data types
* Check Numeric, Categorical Columns
* One-hot encode categorical columns

In [67]:
def explore_data_1(data) :
    print(data.dtypes.value_counts())
    numeric = data._get_numeric_data().columns
    categorical = data.select_dtypes(exclude=["number"]).columns
    print('* Numeric columns : ')
    print(numeric)
    print('* Categorical columns : ')
    print(categorical)
    print('* Num of categories in categorial colums : ')
    print(data.select_dtypes('object').apply(pd.Series.nunique, axis=0))

print('[Product]')
explore_data_1(Product)

print('\n[Search1]')
explore_data_1(Search1)

print('\n[Search2]')
explore_data_1(Search2)

print('\n[Custom]')
explore_data_1(Custom)

print('\n[Session]')
explore_data_1(Session)

print('\n[Master]')
explore_data_1(Master)

#df = pd.get_dummies(df)
#print('Training Features Shape :', df.shape)

[Product]
object    4
int64     4
dtype: int64
* Numeric columns : 
Index(['CLNT_ID', 'SESS_ID', 'HITS_SEQ', 'PD_C'], dtype='object')
* Categorical columns : 
Index(['PD_ADD_NM', 'PD_BRA_NM', 'PD_BUY_AM', 'PD_BUY_CT'], dtype='object')
* Num of categories in categorial colums : 
PD_ADD_NM    759548
PD_BRA_NM     22716
PD_BUY_AM     17731
PD_BUY_CT       361
dtype: int64

[Search1]
int64     3
object    1
dtype: int64
* Numeric columns : 
Index(['CLNT_ID', 'SESS_ID', 'SEARCH_CNT'], dtype='object')
* Categorical columns : 
Index(['KWD_NM'], dtype='object')
* Num of categories in categorial colums : 
KWD_NM    81539
dtype: int64

[Search2]
object    2
int64     1
dtype: int64
* Numeric columns : 
Index(['SESS_DT'], dtype='object')
* Categorical columns : 
Index(['KWD_NM', 'SEARCH_CNT'], dtype='object')
* Num of categories in categorial colums : 
KWD_NM        84607
SEARCH_CNT     1436
dtype: int64

[Custom]
int64     2
object    1
dtype: int64
* Numeric columns : 
Index(['CLNT_ID', 'CLNT_A

In [4]:
#데이터보기
Product.head()

CLNT_ID   SESS_ID  HITS_SEQ    PD_C                         PD_ADD_NM  \
0  4139680   7605037        12  642112                           색상:워터멜론   
1  4140076  10189797        13  570603  색상:BLK0_(BLK0)BLACK|사이즈:120 / 2개   
2  4142395   6158159        85  179538                         (not set)   
3  4144914   7935714        12  554336      색상:블랙|사이즈:160cm(12~13세) / 1개   
4  4144917   6406509        78  190306                                5개   

  PD_BRA_NM PD_BUY_AM PD_BUY_CT  
0   [바비브라운]    39,000         1  
1       데상트    39,000         2  
2     [아베다]    39,000         1  
3   아디다스 키즈    39,000         1  
4      데코르테    39,000         5

In [5]:
Search1.head()

CLNT_ID  SESS_ID     KWD_NM  SEARCH_CNT
0  5607714  7112876   빌리프 아이크림           6
1  5607714  4090791  프리메라 마스크팩           3
2  5607714  4090791      여성청결제           1
3  5612428  1876482       명품가방           1
4  5612428   658123  콩순이 아이스크림           1

In [6]:
Search2.head()

SESS_DT     KWD_NM SEARCH_CNT
0  20180407      닥스원피스          8
1  20180407  닥터지 브라이트닝          1
2  20180407     달팡 인트랄          2
3  20180407        대상트          1
4  20180407       더블유닷          6

In [7]:
Custom.head()

CLNT_ID CLNT_GENDER  CLNT_AGE
0  4830726           F        30
1  4830874           F        40
2  4830975           F        30
3  4831275           F        30
4  4825325           F        30

In [8]:
Session.head()

CLNT_ID   SESS_ID  SESS_SEQ   SESS_DT  TOT_PAG_VIEW_CT TOT_SESS_HR_V  \
0  5873599   8641867         9  20180509             82.0         1,890   
1  5873599   6616320        21  20180611            105.0         1,604   
2  5873599   5886172        40  20180624             41.0           632   
3  5873884   1050889        15  20180913            160.0         1,035   
4  5874461  10298270         5  20180412             13.0           298   

  DVC_CTG_NM       ZON_NM  CITY_NM  
0     mobile      Daejeon  Daejeon  
1     mobile        Busan    Busan  
2     mobile      Daejeon  Daejeon  
3     mobile  Gyeonggi-do   Anyang  
4     mobile        Seoul    Seoul

In [9]:
Master.head()

PD_C                                              PD_NM   CLAC1_NM  \
0  64382   언더아머 남성 UA HG 아머 모크 LS 1289559-001 - 블랙 / MD[95]      스포츠패션   
1  62282                         여자 플라워덧신 2족선물세트 17403-2set  속옷/양말/홈웨어   
2  61729                       88A2933253배트맨스웨트티 - 블랙 / 130      유아동의류   
3  61537            닥터마틴 아드리안 블랙, 체리레드 - 02_체리레드 / 250mm(6)       패션잡화   
4  58820  여성 그레이 스트라이프 퍼프 소매 블라우스 (128865YQ33) - 회색(앤틱실버...       남성의류   

    CLAC2_NM  CLAC3_NM  
0  남성일반스포츠의류  남성스포츠티셔츠  
1      여성양말류    여성일반양말  
2     유아의류상의  영유아티셔츠/탑  
3        남성화      남성부츠  
4     남성의류상의    남성남방셔츠

In [33]:
Product

CLNT_ID   SESS_ID  HITS_SEQ    PD_C  \
0        4139680   7605037        12  642112   
1        4140076  10189797        13  570603   
2        4142395   6158159        85  179538   
3        4144914   7935714        12  554336   
4        4144917   6406509        78  190306   
5        4147643   4050743        66  149995   
6        4147739   6881895        10  568751   
7        4148106   6537318        21  570603   
8        4148129   3942102         2  755148   
9        4148527   5718021        19  608620   
10       4150259   4098361        90  537513   
11       4153618    729571        42   77900   
12       4154576   4253415        19  591229   
13       4157355   6349580        34  554700   
14       4159738   3278944        42  184776   
15       4160219   9232020        32  368706   
16       4161209   3612192       247  125473   
17       4167192  10832157       189  479110   
18       4169076   4316808        24  296741   
19       4170352   4651820       387  166295   
20       4171030   2287214        19  520151   
21       4173605   9718934        23  636556   
22       4174766   1738770        38  156410   
23       4177155  10750856        22  487558   
24       4178471    675974        36   84340   
25       4180381  10005643       169  571504   
26       4183066   1690031        51   65151   
27       4184092  10784068       141  457359   
28       4184818   4539154         2  167474   
29       4185920   6636436        46  581220   
...          ...       ...       ...     ...   
5024876  4099176   8850749       102  672397   
5024877  4103067  10240533        47  467494   
5024878  4104107   9203001       214  453584   
5024879  4104395   5256411       120  704864   
5024880  4107262   1270060        67  778307   
5024881  4108487   5079745        22  725745   
5024882  4108900   2967180        27  642112   
5024883  4108900   2967180        56  642112   
5024884  4109607     61205        77  489144   
5024885  4110447   6192376       114  391199   
5024886  4111339   2566234        38  323529   
5024887  4114333   9706699        71  489450   
5024888  4115601   5154938        30  642112   
5024889  4119818   4140131        87  139554   
5024890  4121511   1184568        68   40071   
5024891  4121682   8736148        37  399606   
5024892  4121860  10876144        12  577503   
5024893  4123650   6943570        19  829673   
5024894  4123943   5105759        11  786487   
5024895  4126622  10737750        15  716942   
5024896  4127373   8591232        28  401577   
5024897  4127769   8541560        95  571188   
5024898  4128823   5004578        33  414556   
5024899  4131574   8663581         5  489577   
5024900  4131835   1060339       145   53273   
5024901  4132078  10042817        76  527736   
5024902  4133768   7272433        31  284643   
5024903  4133768   7946785        38  284643   
5024904  4134952   5355187       113  840639   
5024905  4136808   8930953        33  442584   

                                   PD_ADD_NM     PD_BRA_NM PD_BUY_AM PD_BUY_CT  
0                                    색상:워터멜론       [바비브라운]    39,000         1  
1           색상:BLK0_(BLK0)BLACK|사이즈:120 / 2개           데상트    39,000         2  
2                                  (not set)         [아베다]    39,000         1  
3               색상:블랙|사이즈:160cm(12~13세) / 1개       아디다스 키즈    39,000         1  
4                                         5개          데코르테    39,000         5  
5                     사이즈:옵션04.AN868_04 / 1개        앵글로매니아    39,000         1  
6                      신발색상:블랙(BK)/신발사이즈:280       [디스커버리]    39,000         1  
7           색상:BLK0_(BLK0)BLACK|사이즈:115 / 1개           데상트    39,000         1  
8                              모델명:올나이터 / 1개         어반디케이    39,000         1  
9                                 사이즈:7 / 1개           나이키    39,000         1  
10                            색상:BLK/사이즈:105         [아레나]    39,000         1  
11                   품번:3_JI9DSN12-MB/사이즈:29         [잠뱅이]  

In [34]:
Product_new = Product[Product.HITS_SEQ>1]

In [35]:
Product_new

CLNT_ID   SESS_ID  HITS_SEQ    PD_C  \
0        4139680   7605037        12  642112   
1        4140076  10189797        13  570603   
2        4142395   6158159        85  179538   
3        4144914   7935714        12  554336   
4        4144917   6406509        78  190306   
5        4147643   4050743        66  149995   
6        4147739   6881895        10  568751   
7        4148106   6537318        21  570603   
8        4148129   3942102         2  755148   
9        4148527   5718021        19  608620   
10       4150259   4098361        90  537513   
11       4153618    729571        42   77900   
12       4154576   4253415        19  591229   
13       4157355   6349580        34  554700   
14       4159738   3278944        42  184776   
15       4160219   9232020        32  368706   
16       4161209   3612192       247  125473   
17       4167192  10832157       189  479110   
18       4169076   4316808        24  296741   
19       4170352   4651820       387  166295   
20       4171030   2287214        19  520151   
21       4173605   9718934        23  636556   
22       4174766   1738770        38  156410   
23       4177155  10750856        22  487558   
24       4178471    675974        36   84340   
25       4180381  10005643       169  571504   
26       4183066   1690031        51   65151   
27       4184092  10784068       141  457359   
28       4184818   4539154         2  167474   
29       4185920   6636436        46  581220   
...          ...       ...       ...     ...   
5024876  4099176   8850749       102  672397   
5024877  4103067  10240533        47  467494   
5024878  4104107   9203001       214  453584   
5024879  4104395   5256411       120  704864   
5024880  4107262   1270060        67  778307   
5024881  4108487   5079745        22  725745   
5024882  4108900   2967180        27  642112   
5024883  4108900   2967180        56  642112   
5024884  4109607     61205        77  489144   
5024885  4110447   6192376       114  391199   
5024886  4111339   2566234        38  323529   
5024887  4114333   9706699        71  489450   
5024888  4115601   5154938        30  642112   
5024889  4119818   4140131        87  139554   
5024890  4121511   1184568        68   40071   
5024891  4121682   8736148        37  399606   
5024892  4121860  10876144        12  577503   
5024893  4123650   6943570        19  829673   
5024894  4123943   5105759        11  786487   
5024895  4126622  10737750        15  716942   
5024896  4127373   8591232        28  401577   
5024897  4127769   8541560        95  571188   
5024898  4128823   5004578        33  414556   
5024899  4131574   8663581         5  489577   
5024900  4131835   1060339       145   53273   
5024901  4132078  10042817        76  527736   
5024902  4133768   7272433        31  284643   
5024903  4133768   7946785        38  284643   
5024904  4134952   5355187       113  840639   
5024905  4136808   8930953        33  442584   

                                   PD_ADD_NM     PD_BRA_NM PD_BUY_AM PD_BUY_CT  
0                                    색상:워터멜론       [바비브라운]    39,000         1  
1           색상:BLK0_(BLK0)BLACK|사이즈:120 / 2개           데상트    39,000         2  
2                                  (not set)         [아베다]    39,000         1  
3               색상:블랙|사이즈:160cm(12~13세) / 1개       아디다스 키즈    39,000         1  
4                                         5개          데코르테    39,000         5  
5                     사이즈:옵션04.AN868_04 / 1개        앵글로매니아    39,000         1  
6                      신발색상:블랙(BK)/신발사이즈:280       [디스커버리]    39,000         1  
7           색상:BLK0_(BLK0)BLACK|사이즈:115 / 1개           데상트    39,000         1  
8                              모델명:올나이터 / 1개         어반디케이    39,000         1  
9                                 사이즈:7 / 1개           나이키    39,000         1  
10                            색상:BLK/사이즈:105         [아레나]    39,000         1  
11                   품번:3_JI9DSN12-MB/사이즈:29         [잠뱅이]  

# Merge Data
1) Merge product purchase data to see features which is expected to affect on purchase of client bought given product group, not individual product

- Merge following csv files
    * Product.csv : CLNT_ID, SESS_ID(Brdige), PD_C(Bridge), PD_BUY_CT, PD_BUY_AM
    * Master.csv : PD_C(Bridge), CLAC1_NM, CLAC2_NM, CLAC3_NM
    * Session.csv : SESS_ID(Bridge), SESS_DT, DVC_CTG_NM, ZON_NM, CITY_NM
    * Custom.csv : CLNT_ID, CLNT_GENDER, CLNT_AGE
    
    
- To get data below
  
  
- Client Information
    * CLNT_ID : Client ID (Primary Key)
    * CLNT_GENDER : Client Gender
    * CLNT_AGE : Client Age
    * DVC_CTG_NM : Client Device
    * ZON_NM : Client Location (briefly)
    * CITY_NM : Client Location (in detail)
    
- Purchase Information
    * SESS_DT : Session Date
    * PD_BUY_CT : Number of product purchased
    * PD_BUY_AM : Price of product purchased
    
- Product Information
    * CLAC1_NM : Product group (largest)
    * CLAC2_NM : Product group (moderate)
    * CLAC3_NM : Product group (smallest)
    
    
### Now we will try to decrease the size of the file by Merging them together
    
  

In [81]:
#구매데이터의 client 수 != 고객데이터의 client 수 
#구매데이터의 client 수 != 검색데이터의 client 수

print('Product shape: ',Product.shape)
print('Custom shape: ',Custom.shape)
print('Search1 shape: ',Search1.shape)
print('Session shape: ',Session.shape)
print()
print('구매에 참여한 client 수: ',len(Product['CLNT_ID'].value_counts()))
print('고객데이터에 있는 unique한 client 수: ',len(Custom['CLNT_ID'].value_counts()))
print('세션에 참여한 unique한 client 수: ',len(Session['CLNT_ID'].value_counts()))
print('검색에 참여한 client 수: ', len(Search1['CLNT_ID'].value_counts()))

Product shape:  (5024906, 8)
Custom shape:  (671679, 3)
Search1 shape:  (2884943, 4)
Session shape:  (2712907, 9)

구매에 참여한 client 수:  922737
고객데이터에 있는 unique한 client 수:  671679
세션에 참여한 unique한 client 수:  922737
검색에 참여한 client 수:  511477


- 세션에 참여한 client 수와 구매에 참여한 client 수는 동일하다
- 상품을 구매했을지라도 검색을 하지 않은 client들이 있다
- 922737 - 511477 = 411260
- 고객데이터에 모든 client들의 정보가 있는 것은 아니다

In [74]:
temp1 = pd.merge(Search1, Product, on = ['CLNT_ID', 'SESS_ID']) 

In [75]:
temp1.head(1)

CLNT_ID  SESS_ID    KWD_NM  SEARCH_CNT  HITS_SEQ    PD_C   PD_ADD_NM  \
0  5607714  7112876  빌리프 아이크림           6       109  656297  기본:기본 / 1개   

  PD_BRA_NM PD_BUY_AM PD_BUY_CT  
0       빌리프    48,000         1

In [76]:
print(len(temp1['CLNT_ID'].value_counts()))

511477


In [77]:
temp2 = pd.merge(Custom, Product, on = 'CLNT_ID')

In [78]:
print(len(temp1['CLNT_ID'].value_counts()))

511477


In [79]:
temp3 = pd.merge(Search1, Product_new, on = ['CLNT_ID', 'SESS_ID']) 

In [80]:
print(len(temp3['CLNT_ID'].value_counts()))

511447


- 전처리한 데이터가 클라이언트에 영향을 끼치지 않았음을 볼수 있다

In [82]:
print(len(Product_new['CLNT_ID'].value_counts())) 

922618


In [83]:
print(len(Product['CLNT_ID'].value_counts())) 

922737


In [84]:
##922737->922618
##전처리된 값으로 변경 (HIT_SEQ가 1인 것 제거하기)
Product = Product_new

- Search1의 client < custom의 client < Product의 client = Session의 Client
Search1의 client 중에 custom 정보 없는 것 있다.
검색을 하고, custom 정보가 있는 client만을 기준으로 table 작성 (375864개의 client id)

In [88]:
#Search1의 client 중에 custom 정보 없는 것이 있다. 
temp = pd.merge(Custom, Search1, how = 'inner', on = 'CLNT_ID')
print(len(temp['CLNT_ID'].value_counts()))

375864


### Product를 기준으로 Merging 시작
  ### data1
    1) Product(Hit수를 제거한) + Session

In [89]:
#Product에서 outlier를 제거해 줬으므로 Product의 client를 기준으로 묶는다.
data1 = pd.merge(Product, Session)
print(len(data1['CLNT_ID'].value_counts()))
print('아주 조금 줄었음')

922618
아주 조금 줄었음


### data2
    2) 상품정보 추가  + Master

In [93]:
data2 = pd.merge(data1, Master, how = 'left', on = 'PD_C')
print(len(data1['CLNT_ID'].value_counts()))
print('변동없음')

922618
변동없음


### Client_Behavior
   1) Search1 과 Customer사이의 공통군 추출

In [114]:
# Search1 과 Customer사이의 공통군 추출
cust_search =  pd.merge(Search1, Custom)

In [119]:
print(len(Search1['CLNT_ID'].value_counts()))

511477


In [120]:
print(len(Custom['CLNT_ID'].value_counts()))

671679


In [121]:
print(len(cust_search['CLNT_ID'].value_counts()))

375864


In [122]:
cust_search

CLNT_ID   SESS_ID         KWD_NM  SEARCH_CNT CLNT_GENDER  CLNT_AGE
0        5607714   7112876       빌리프 아이크림           6           F        40
1        5607714   4090791      프리메라 마스크팩           3           F        40
2        5607714   4090791          여성청결제           1           F        40
3        5607714  10335152           프리메라           3           F        40
4        5607714   6476370        헤어 드라이어           3           F        40
5        5607714   6476370      대용량에어프라이어           2           F        40
6        5607714   9507944           프리메라           2           F        40
7        5607714  10335152            빌리프           3           F        40
8        5607714   7112876         헤라 베이스           1           F        40
9        5607714   7112876             헤라           1           F        40
10       5607714   6476370         에어프라이어           5           F        40
11       5607714   4090791          여성보드숏           1           F        40
12       5607714   4090791          바디미스트           2           F        40
13       5607714   4090791        원피스래쉬가드           2           F        40
14       5607714   4090791            빌리프           6           F        40
15       5607714   9507944        프리메라 필링           4           F        40
16       5607714  10335152  빌리프 수분크림 75ml           1           F        40
17       5607714   6476370            앳코너           1           F        40
18       5607714   8361954        남성수영복바지           3           F        40
19       5607714   8361954         남성 보드숏           5           F        40
20       5612428   1876482           명품가방           1           F        30
21       5612428    658123      콩순이 아이스크림           1           F        30
22       5612428   1876482     블루독 티셔츠 남아           1           F        30
23       5613055   8510027         임산부 팬티           2           F        30
24       5613055   8510027        임산부 원피스           1           F        30
25       5609603  10483281          베베드피노          27           F        30
26       5609603  10483281            쁘띠뮤           6           F        30
27       5609603  10483281      베베드피노 티셔츠           2           F        30
28       5609763   6286724      남성 반팔 티셔츠           4           M        30
29       5609800   7380789       메트로시티 지갑           1           F        60
...          ...       ...            ...         ...         ...       ...
2297694  6387717   1878814            메리즈           2           F        30
2297695  6387734     31646      네스카페돌체구스토           1           M        50
2297696  6387734    150322          여아티셔츠           1           M        50
2297697  6387734    376589    아디다스키즈 수퍼스타           1           M        50
2297698  6387734    150322       여아주니어후드티           2           M        50
2297699  6387734    376589         아디다스키즈           1           M        50
2297700  6387880   3634527       디스커버리반바지           6           F        40
2297701  6387880   3547761            글리터           4           F        40
2297702  6387880   3844161     캐네스레이디 원피스           1           F        40
2297703  6387880   3634527        스파이더반바지           2           F        40
2297704  6387880   3844161     케네스레이디 원피스           6           F        40
2297705  6387880   3844161         진주 원피스           4           F        40
2297706  6387880   3547761        데상트 반바지           2           F        40
2297707  6387880   3547761           맥 대즐           5           F        40
2297708  6388296   2914718        엠폴햄 티셔츠           1           F        40
2297709  6388296   2914718           폴햄셔츠           2           F        40
2297710  6388296   2914718            펠틱스           1           F        40
2297711  6388296   2914718             폴햄           1           F        40
2297712  6388327   8729180           에티튜드           1           F        20
2297713  6388327   8729180      에티튜드 세탁세제           1           F        20
2297714  6382387   7858219           보드게임           3    

In [123]:
#Search1의 client기준으로 병합
#client -> 검색 -> 구매
Client_Behavior = pd.merge(data2, cust_search, on='CLNT_ID')

In [124]:
print(len(Client_Behavior['CLNT_ID'].value_counts()))
print('줄었음')

375864
줄었음


In [107]:
Client_Behavior.head()

CLNT_ID  SESS_ID  HITS_SEQ    PD_C             PD_ADD_NM PD_BRA_NM  \
0  4142395  6158159        85  179538             (not set)     [아베다]   
1  4142395  6158159        85  179538             (not set)     [아베다]   
2  4142395  6158159        85  179538             (not set)     [아베다]   
3  4142395  6158159        85  179538             (not set)     [아베다]   
4  4142395  6158159        23  558739  상품명:이솝 제라늄 리프 바디 클렌저      [이솝]   

  PD_BUY_AM PD_BUY_CT  SESS_SEQ   SESS_DT    ...      TOT_SESS_HR_V  \
0    39,000         1         1  20180619    ...              6,524   
1    39,000         1         1  20180619    ...              6,524   
2    39,000         1         1  20180619    ...              6,524   
3    39,000         1         1  20180619    ...              6,524   
4    56,000         1         1  20180619    ...              6,524   

  DVC_CTG_NM ZON_NM CITY_NM                                      PD_NM  \
0    desktop  Seoul   Seoul                [단독][6월] 라이트 엘리먼츠 헤어 에센스 세트   
1    desktop  Seoul   Seoul                [단독][6월] 라이트 엘리먼츠 헤어 에센스 세트   
2    desktop  Seoul   Seoul                [단독][6월] 라이트 엘리먼츠 헤어 에센스 세트   
3    desktop  Seoul   Seoul                [단독][6월] 라이트 엘리먼츠 헤어 에센스 세트   
4    desktop  Seoul   Seoul  이솝 제라늄 리프 바디 클렌저 500ml - 이솝 제라늄 리프 바디 클렌저   

  CLAC1_NM CLAC2_NM  CLAC3_NM   KWD_NM SEARCH_CNT  
0    퍼스널케어     헤어케어  헤어케어선물세트     입생로랑          3  
1    퍼스널케어     헤어케어  헤어케어선물세트       이솝          3  
2    퍼스널케어     헤어케어  헤어케어선물세트  아베다 스무드          1  
3    퍼스널케어     헤어케어  헤어케어선물세트      아베다         23  
4    퍼스널케어     바디케어      바디워시     입생로랑          3  

[5 rows x 21 columns]

In [108]:
#Client_Behavior 데이터 내보내기
Client_Behavior.to_csv("C:/Users/ASUS/Desktop/KuPoint_Project/Client_Behavior.csv", encoding="UTF-8")

### 상품 구매량 count를 위한 테이블 만들기
- 상품 구매 count를 하기 위해서, Product에서 HIT_SEQ = 1을 제거하고(앞에서 이미 제거됨), Search1과 custom에 둘 다 존재하는 Client 기준으로 정렬
- 한 client의 hit번호가 가장 큰 것에 대해서만 따로 빼서 매트릭스를 만들고, 그 후 상품분류에 대해 value_counts()해 주면 될듯

In [99]:
#Search1의 client를 unique하게 만들어주기 
Client = pd.merge(Search1[['CLNT_ID']], Custom[['CLNT_ID']])
Unique_Client = Client.drop_duplicates(subset=None, keep='last', inplace=False)

In [100]:
Unique_Client['CLNT_ID'].value_counts()

5461714    1
557637     1
5280327    1
5328413    1
2685514    1
6353483    1
4246092    1
4768333    1
55886      1
2150991    1
1583696    1
1581649    1
5767765    1
5773910    1
2365283    1
1079898    1
553563     1
5784157    1
6314590    1
6400608    1
2380969    1
2136646    1
3191363    1
4842089    1
2772219    1
4954662    1
2855463    1
2353704    1
1303081    1
5857480    1
          ..
4580630    1
2987288    1
2465049    1
4762753    1
6130971    1
1081954    1
4589347    1
5615932    1
4015362    1
4740254    1
1896362    1
1211621    1
6448358    1
1532795    1
5179014    1
3592504    1
4859118    1
3331312    1
5954801    1
6487285    1
5141990    1
4908278    1
6505990    1
2266360    1
5938425    1
6448371    1
4371708    1
3319038    1
6466815    1
6291456    1
Name: CLNT_ID, Length: 375864, dtype: int64

In [109]:
#Search1과 custom에 공통으로 있는 client id에 해당하는 구매내역만 dataframe으로 만들기 
Product_Client = pd.merge(Product, Unique_Client, how = 'inner', on = 'CLNT_ID')
print(len(Product_Client['CLNT_ID'].value_counts()))

375864


In [110]:
Product_Client = Product_Client.drop_duplicates(subset=None, keep='last', inplace=False)
print(len(Product_Client['CLNT_ID'].value_counts()))

375864


In [111]:
Product_Client

CLNT_ID   SESS_ID  HITS_SEQ    PD_C  \
0        4140076  10189797        13  570603   
1        4140076  10342512        23  422358   
2        4140076   6526651        22  196272   
3        4140076   8326529       199  305965   
4        4140076   8891563        19  305965   
5        4140076   6526651       193  196272   
6        4140076   6591360       149  196272   
7        4140076   6526651        22  365025   
8        4140076   4756252       147  794908   
9        4140076  10342512        23  403243   
10       4140076   9147546       474  544957   
11       4140076   6526651        22  364992   
12       4140076   6591360       149  628713   
13       4140076  10574005       253  450131   
14       4140076   9147546       462  533783   
15       4140076   6526651        22  339321   
16       4140076  10767546        19  600758   
17       4140076  10767546        19  471477   
18       4140076   7754761        71  315988   
19       4140076   6517872       179  373461   
20       4140076   5338950        76  598881   
21       4140076  10883079        29  689663   
22       4140076  10584304        50  689663   
23       4140076  10760242        26  689663   
24       4140076   6885554       134  288192   
25       4140076   6885554       118  522552   
26       4140076   9992238       112  495110   
27       4140076   8153697       123  555694   
28       4140076   6591360       149  526369   
29       4140076  10342512        23  806382   
...          ...       ...       ...     ...   
3073269  5120355   6897421        38  692796   
3073270  5120355   6900636        16  692796   
3073271  1835621   2942074        47  250634   
3073272  1835621   3242152        60  250634   
3073273  5010268   8741152        33  636556   
3073274  5010268   8699707        29  636556   
3073275  5010268   8699707        63  636556   
3073276  1082242   5661700        58  179401   
3073277   372204   5634118        48  403151   
3073278  3650646  10447349        33  563577   
3073279  5033434   4390462        75  507859   
3073280  3853640  10507265        54  687956   
3073281  4809147   7145157        63  224460   
3073282  4809147   7186972        47  224460   
3073283  5601778   8217108       130  423262   
3073284  5601778   8217108        91  423262   
3073285   901964      8750       285   15477   
3073286  1454857   9954443        26  816627   
3073287  1454857   9953776        23  816627   
3073288  1337534  10680863        53  419179   
3073289  4110447   5212278        90  636556   
3073290  4110447   6192376       114  391199   
3073291  4926545   9037714        54  455563   
3073292  5072758   8571308        27  819710   
3073293  2644758   6801437        83  376829   
3073294  2644758   6806773        21  388660   
3073295  5085969  10609581        48  590876   
3073296  3235159  10660304        94    3253   
3073297  4108900   2967180        27  642112   
3073298  4108900   2967180        56  642112   

                                   PD_ADD_NM     PD_BRA_NM PD_BUY_AM PD_BUY_CT  
0           색상:BLK0_(BLK0)BLACK|사이즈:120 / 2개           데상트    39,000         2  
1        선택:02.204143-02S|사이즈:M5W7(240) / 1개           크록스    39,900         1  
2            색상:선택3)248(네이비)|사이즈:L(105) / 1개         프레드페리    47,600         1  
3                           사이즈:115_3XL / 1개      아디다스(의류)    46,700         1  
4                           사이즈:115_3XL / 1개      아디다스(의류)    46,700         1  
5            색상:선택2)102(블랙)|사이즈:XL(110) / 1개         프레드페리    47,600         1  
6            색상:선택2)102(블랙)|사이즈:XL(110) / 2개         프레드페리    47,600         2  
7                         색상:WH|사이즈:165 / 1개         닥스 키즈    48,300         1  
8                                         1개         네이쳐리빙     7,200         1  
9                                사이즈:M9 / 1개           크록스    57,000         1  
10                                        1개           영실업    59,820         1  
11                        색상:BG|사이즈:165 / 1개         닥스 키즈  

### Explore Data
    - Brief visualization

In [ ]:
heavy_light_users = purchase['CLNT_ID'].value_counts()
print(heavy_light_users)